### DISCUSSION TITLE
Remove disk name filter for Virtual Machine Image scan

### DISCUSSION THREAD

**Discussion by yusuke-koyoshi** [aquasecurity/trivy#5450] [Ideas]
`trivy vm` only scans specific disk names for snapshots. 
https://github.com/aquasecurity/trivy/blob/3be5e6b24271a21f5aa700bab7cc32d1549b8199/pkg/fanal/walker/vm.go#L22-L30

OS packages are located in the root volume, but programming language project files may exist outside of the root volume, so it is not advisable to limit them to specific volume names.

**Comment by knqyf263** [comment]
@masahiro331 Wdyt?

**Comment by masahiro331** [comment]
@yusuke-koyoshi
If they are divided by logical volumes, which I believe is the case, but are they divided by physical volumes?

**Reply by masahiro331** [reply]
At least an AMI is composed of a single physical volume.

**Reply by yusuke-koyoshi** [reply]
@masahiro331 
An AMI consists of a single physical volume, but multiple EBS volumes can be attached to EC2.
When you scan EC2 instance, you must scan all attached EBS volumes.

**Reply by yusuke-koyoshi** [reply]
@masahiro331 
There can be cases where multiple EBS volumes are attached and the partition names are something other than defined.
I don't think there is a need to restrict partition names, but why do we do so?

**Reply by masahiro331** [reply]
Since there are hard disks for backup and journaling, the policy was to ignore volumes that are not known.
